src/eredis_sub_client.erl

```diff
@@ -197,9 +236,10 @@ handle_info({reconnect_failed, Reason}, State) ->
 
 %% Redis is ready to accept requests, the given Socket is a socket
 %% already connected and authenticated.
-handle_info({connection_ready, Socket}, #state{socket = undefined} = State) ->
+handle_info({connection_ready, Socket},
```

Comment:
As a note, when a re-connection occur it looks like we don't re-subscribe to the channels we setup before i.e. the ones we have in `state{channels}`.
This seemed to be the users responsibility in the legacy code, probably by re-subscribing when receiving `eredis_connected`.

Reply:
Interesting. Let's consider that a separate bug and open an issue for it.

Reply:
#50